lfsapi/auth.go

```diff
@@ -24,8 +24,22 @@ var (
 // DoWithAuth sends an HTTP request to get an HTTP response. It attempts to add
 // authentication from netrc or git's credential helpers if necessary,
 // supporting basic and ntlm authentication.
-func (c *Client) DoWithAuth(remote string, access Access, req *http.Request) (*http.Response, error) {
-	return c.doWithAuth(remote, access, req, nil)
+func (c *Client) DoWithAuth(remote string, access Access, req *http.Request, allowRetry bool) (*http.Response, error) {
```

Comment:
Nice, I think that your explanation as to why we need this new behavior variant makes sense to me. What do you think about introducing another function which encodes this behavior, rather than adding a boolean argument to this one?

This has a few of benefits in my mind: (1) is that it limits the number of parameters that we have to think about when calling a function, (2) it makes clear at the call-site what the parameter is doing (i.e., rather than seeing `..., true` in a function call, we get a name that describes the behavior), and (3) it makes explicit what's going on in (2) which is really that we have two different versions of this function.

Reply:
I like that idea! I actually considered it but went with the boolean because all the names I came up with at the time felt pretty unwieldy. I actually quite like the name `DoWithAuthNoRetry()` though, so I'll add that.